### ISSUE TITLE
Step 5 Gross Monthly Income: Fix accessibility issue/keyboard navigation for Gross Monthly Income list loop

### ISSUE THREAD

**Issue by vbellissimo** [department-of-veterans-affairs/va.gov-team#73299]
## Issue Description

Slack thread: https://dsva.slack.com/archives/C05MS96P6P2/p1704836391539569?thread_ts=1704484692.155699&cid=C05MS96P6P2

When you add another income source keyboard focus goes to "Add another income source" button.

Expected behavior: Keyboard focus goes to New income source so I can hear the new section and tab to the first new field

<img width="536" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBNUN4QWc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--a03d1dd3dd04530a3f9b5ca71bf1d4ef1eaf6518/Screenshot%202024-01-09%20at%207.05.24%E2%80%AFPM.png" alt="Screenshot 2024-01-09 at 7.05.24 PM.png" />

## Acceptance Criteria
- [ ] If the user adds more than 1 income source, they keyboard focus goes to New income source so I can hear the new section and tab to the first new field (not the Add another income source" button).

**Comment by gmaciolek** [comment]
I think the AC for this should be "If the user adds more than 1 income source, they keyboard focus goes to the first group of inputs in the New income section, and the input is announced by a screenreader" Keyboard focus only ever goes to interactive elements (not static content like headings).

If we update the AC, Todd's fix resolves the issue. I validated on staging that, after pressing the "Add another income source" button, the keyboard focus goes to the first radio button in the “What type of income?” question in the New income source section. That’s also what’s being announced in VoiceOver on Mac.
